circle.yml

```diff
@@ -9,6 +9,11 @@ checkout:
   post:
     # Update the `master` branch. Otherwise CircleCI will give us a cached one.
     - git fetch -u origin +master:master
+    # Checkout the merged PR for testing as CircleCI will just use the PR head otherwise.
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git fetch origin +refs/pull/$CIRCLE_PR_NUMBER/head:pr/$CIRCLE_PR_NUMBER/head ; fi
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git fetch origin +refs/pull/$CIRCLE_PR_NUMBER/merge:pr/$CIRCLE_PR_NUMBER/merge ; fi
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git checkout -qf pr/$CIRCLE_PR_NUMBER/merge ; fi
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git branch --merged | grep master > /dev/null ; fi
```

Comment:
Make sure `master` is merged into the PR merge commit. If it is not, that means we have a merge conflict with `master`. So error out if that is the case.